### PR TITLE
Update nodeSelector to use built-in role label

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

File: submit.yml.erb [hunk 1]
```diff
@@ -53,7 +53,7 @@ script:
       cpu: "<%= container_sizes[size.to_sym][:cpu] %>m"
       memory: "<%= container_sizes[size.to_sym][:memory] %>Gi"
     node_selector:
-      osc.edu/role: ondemand
+      node-role.kubernetes.io/ondemand: ''
```
Comment (johrstrom) [inline submit.yml.erb:56]: That seems like odd syntax, all you need is the key?

Comment (treydock) [reply submit.yml.erb:56]: Yep only the key matters.  It's a label with special meaning, the part after `node-role.kubernetes.io/` is the actual role value that shows up in things like `kubectl get nodes`.

```
]# kubectl get nodes
NAME             STATUS   ROLES                        AGE    VERSION
kubecontroller   Ready    control-plane,master         364d   v1.23.6
kubeworker01     Ready    ondemand                     364d   v1.23.6
kubeworker02     Ready    ondemand                     364d   v1.23.6
kubeworker03     Ready    ondemand                     364d   v1.23.6
kubeworker04     Ready    infrastructure,webservices   363d   v1.23.6
kubeworker05     Ready    infrastructure,webservices   363d   v1.23.6
kubeworker06     Ready    infrastructure,webservices   317d   v1.23.6
kubeworker07     Ready    infrastructure,webservices   317d   v1.23.6
kubeworker08     Ready    ondemand                     268d   v1.23.6
kubeworker09     Ready    ondemand                     155d   v1.23.6
kubeworker10     Ready    ondemand                     155d   v1.23.6
```

The value of label can be anything, it's not used for anything.  This is something Kubernetes does out of the box for the controller nodes:

```
[root@kubecontroller ~]# kubectl get node kubecontroller -o yaml | grep node-role
    node-role.kubernetes.io/control-plane: ""
    node-role.kubernetes.io/master: ""
```

This particular label also has extra security around it, ie I can only set the value with ClusterAdmin role, where as the other labels a kubelet can set themselves.

**Review by johrstrom — APPROVED** [review]

**Comment by treydock** [comment]
@johrstrom Be good to get this in tag that goes out with next release.

**Comment by johrstrom** [comment]
Sorry, didn't realize this was unreleased. You're right, that we should have 0.4.1 with this. @gerald-byrket or @Oglopf can you take care?

**Pull request merged**